src/config.ts

```diff
@@ -132,20 +107,15 @@ export class GraphQLConfig {
   readonly filepath: string;
   readonly dirpath: string;
 
-  readonly projects: {
-    [name: string]: GraphQLProjectConfig;
-  };
+  readonly projects: Record<string, GraphQLProjectConfig> = Object.create(null);
```

Comment:
@B2o5T @kamilkisiela 
`Object.create(null)` causes `TypeError: config.projects.hasOwnProperty is not a function`.

```ts
try {
  for (const projectName in config.projects) {
    if (config.projects.hasOwnProperty(projectName)) {
      const project = config.projects[projectName];

      if (project.hasExtension('codegen')) {
        return true;
      }
    }
  }
} catch (e) {}
```
https://github.com/dotansimha/graphql-code-generator/blob/3c6abbde7a20515d9a1d55b4003ef365d248efb5/packages/graphql-codegen-cli/src/graphql-config.ts#L62-L72

This means a GraphQL config file which has `projects` without `default` project fails to generate.

Reply:
@puku0x thanks for reporting it, I’ll fix it today, and will add a test for the future 👍

Reply:
@puku0x fixed in [`graphql-config@4.3.0`](https://github.com/kamilkisiela/graphql-config/releases/tag/v4.3.0)